pkg/storage/cache/cache.go

```diff
@@ -21,20 +28,29 @@ type Cache struct {
 	metrics metrics
 }
 
-// Wrap adds a layer of in-memory caching to basic Store operations.
-// This call will panic if the capacity is less than or equal to zero.
-// It will also panic if the given store implements storage.Tx.
-func Wrap(store storage.Store, capacity int) *Cache {
+// Wrap adds a layer of in-memory caching to storage.Reader Get and Has operations.
+// It returns an error if the capacity is less than or equal to zero or if the
+// given store implements storage.Tx
+func Wrap(store storage.Store, capacity int) (*Cache, error) {
 	if _, ok := store.(storage.Tx); ok {
-		panic(errors.New("cache should not be used with transactions"))
+		return nil, errors.New("cache should not be used with transactions")
 	}
 
 	lru, err := lru.New[string, []byte](capacity)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
-	return &Cache{store, lru, newMetrics()}
+	return &Cache{store, lru, newMetrics()}, nil
+}
+
+// MustWrap is like Wrap but panics on error.
+func MustWrap(store storage.Store, capacity int) *Cache {
+	c, err := Wrap(store, capacity)
+	if err != nil {
+		panic(err)
```

Comment:
is it ok if we panic while we init the state store?

Reply:
Good catch, fixed!